### PR TITLE
Fix issue related to generating clients when relation annotations are empty

### PR DESCRIPTION
**Pull request by sahanHe**
## Purpose

Fixes: [#3594](https://github.com/ballerina-platform/ballerina-standard-library/issues/3594)

## Checklist
- [x] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_18/entities.bal [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    @persist:AutoIncrement
+    readonly int itemId = -1;
+
+    string name;
+    string 'type;
+    string unit;
+};
+
+
+
```
Comment (daneshk) [inline persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_18/entities.bal:49]:
```suggestion
```

File: persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_18/main.bal [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import ballerina/io;
```
Comment (daneshk) [inline persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_18/main.bal:1]: License headers?

File: persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_19/entities.bal [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    @persist:Relation {keyColumns: ["profileId"]}
+    Profile profile?;
+
+    @persist:Relation
+    User user?;
+|};
+
+
+
```
Comment (daneshk) [inline persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_19/entities.bal:56]:
```suggestion
```

Comment (daneshk) [reply persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_19/entities.bal:56]: remove additional new lines in other places

File: persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_18/main.bal (hunk 2 above)
Comment (daneshk) [reply persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_18/main.bal:1]: check and add missing licenses in other places

File: persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_18/entities.bal (hunk 1 above)
Comment (sahanHe) [reply persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_18/entities.bal]: Updated in [c7709b2](https://github.com/ballerina-platform/persist-tools/pull/48/commits/c7709b28a97d8f91b708b4bd3d483e0a0eebb499)

File: persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_18/main.bal (hunk 2 above)
Comment (sahanHe) [reply persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_18/main.bal:17]: Updated in [c7709b2](https://github.com/ballerina-platform/persist-tools/pull/48/commits/c7709b28a97d8f91b708b4bd3d483e0a0eebb499)

File: persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_19/entities.bal (hunk 3 above)
Comment (sahanHe) [reply persist-cli-tests/src/test/resources/test-src/input/tool_test_generate_19/entities.bal]: Updated in [c7709b2](https://github.com/ballerina-platform/persist-tools/pull/48/commits/c7709b28a97d8f91b708b4bd3d483e0a0eebb499)

File: persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java [hunk 4]
```diff
@@ -1075,6 +1075,27 @@ public static void generateRelations(ArrayList<Entity> entityArray) {
...
+                                    relation.getKeyColumns().add(
+                                            childEntity.getEntityName().toLowerCase() + modifiedKey.substring(0, 1)
+                                                    .toUpperCase() + modifiedKey.substring(1));
+                                }
+                            } else if (relation.getReferences().isEmpty()) {
+                                for (String key : childEntity.getKeys()) {
+                                    String modifiedKey = key.replace("\"", "").trim();
+                                    relation.getReferences().add(modifiedKey);
+                                }
+                            }
```
Comment (daneshk) [inline persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java:1098]: don't we need a else block here?

File: persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java [hunk 5]
```diff
@@ -1158,6 +1179,28 @@ public static void generateRelations(ArrayList<Entity> entityArray) {
...
+                                    childEntity.getRelations().get(relationIndex).getKeyColumns().add(
+                                            entity.getEntityName().toLowerCase() + modifiedKey.substring(0, 1)
+                                                    .toUpperCase() + modifiedKey.substring(1));
+                                }
+                            } else if (childEntity.getRelations().get(relationIndex).getReferences().isEmpty()) {
+                                for (String key : entity.getKeys()) {
+                                    String modifiedKey = key.replace("\"", "").trim();
+                                    childEntity.getRelations().get(relationIndex).getReferences().add(modifiedKey);
+                                }
+                            }
```
Comment (daneshk) [inline persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java:1203]: don't we need a else block here?

File: persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java (hunk 4 above)
Comment (sahanHe) [reply persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java:1098]: Else state will be the case where both annotations are present and this case is the default case where there is no additional processing is required. As we change other states (annotations are empty and half empty) in to this case through the if and else if blocks, the Else block is not required for this instance.

File: persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java (hunk 5 above)
Comment (sahanHe) [reply persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java:1203]: Else state will be the case where both annotations are present and this case is the default case where there is no additional processing is required. As we change other states (annotations are empty and half empty) in to this case through the if and else if blocks, the Else block is not required for this instance.

**Review by daneshk — APPROVED** [review]

**Pull request merged**